google/cloud/storage/internal/object_streambuf.h

```diff
@@ -146,6 +148,7 @@ class ObjectWriteStreambuf : public std::basic_streambuf<char> {
   HashValidator::Result hash_validator_result_;
 
   StatusOr<HttpResponse> last_response_;
+  Status last_status_;
```

Comment:
Is there some relationship between last_response_ and last_statsus_? Like, if last_status_ is OK, then last_response_ is some 200 status? Or if last_response_ is not a 2xx HTTP status, then last_status_ is not OK?

It looks like that might be the case, other than in the constructor where last_response_ is initialized to a 400 HTTP status code, and last_status_ is initialized with OK.

I'm not suggesting any change here, just trying to understand this better since I'm not too familiar with this code.

Reply:
I think this is the key question.

For background, the application usually uses a `gcs::ObjectWriteStream`, and it performs error handling using IOStream errors, for example:

```C++
  gcs::ObjectWriteStream stream = client.WriteObject(....);
  ...
  ..
  ..
  .
  if (!stream.good()) {
   std::cerr << "Error, the last status was = " << stream.last_status(); // TODO: this is the point of this PR.
  }
```

I think we want that error checking code to always work, regardless of the previous operations (`.write()`, `.Close()`, `operator<<`, whatever). I think what we should do is change the code to be something like:

```
Status last_status() const {
  if (last_response_) {
    return Status();
  }
  return last_response_.status();
}
```

And always set `last_reponse_` when there is an error.

Reply:
done.